### PR TITLE
TextField: template with label and icon

### PR DESCRIPTION
**Pull request by TomaszLukawskiSam**
[Issue] https://github.com/Samsung/TAU/issues/1587
[Problem] TextField with label + icon is not implemented
[Solution]
     - new template has been added
     - added new styles

[Screenshot]
image

### CODE REVIEW

File: src/css/profile/mobile/common/textinput.less [hunk 1]
```diff
@@ -316,8 +316,38 @@ input.ui-text-input + .ui-text-input-has-label {
...
+	flex-direction: column;
+	margin: 0 24 * @px_base;
+	flex: 1;
+}
+.ui-text-input-icon {
+	width: 20 * @px_base;
+	height: 20 * @px_base;
+    position: inherit;
+    display: flex;
+    margin: auto 0 auto 24px;
```
Comment (HunseopJeong) [inline src/css/profile/mobile/common/textinput.less]: Fix indentation

File: src/js/profile/mobile/widget/TextInput.js [hunk 2]
```diff
@@ -448,21 +519,26 @@
 					ui = self._ui,
 					btn = ui.textClearButtonElement;
 
-				if (element.value === "" && btn) {
+				self._state.empty =	element.value === "";
```
Comment (HunseopJeong) [inline src/js/profile/mobile/widget/TextInput.js]: ditto.

File: src/js/profile/mobile/widget/TextInput.js [hunk 3]
```diff
@@ -396,6 +414,64 @@
...
+
+			/**
+			 * Find icon assigned to text input
+			 * @method _findIcon
+			 * @param {HTMLElement} element
+			 * @protected
+			 * @member ns.widget.mobile.TextInput
+			 */
+			prototype._findIcon = function (element) {
+				var	parentElement = element.parentElement,
```
Comment (HunseopJeong) [inline src/js/profile/mobile/widget/TextInput.js]: ditto

File: src/js/profile/mobile/widget/TextInput.js [hunk 4]
```diff
@@ -254,6 +260,18 @@
...
+					/**
+					 * Set class for icon assigned to text input
+					 * @style ui-text-input-icon
+					 * @member ns.widget.mobile.TextInput
+					 */
+					ICON: CLASSES_PREFIX + "-icon",
+					/**
+					 * Set class for icon assigned to text input when TextInput is not empty and valid
+					 * @style ui-text-input-not-empty
+					 * @member ns.widget.mobile.TextInput
```
Comment (HunseopJeong) [inline src/js/profile/mobile/widget/TextInput.js:272]: It seems that the selector is different from the defined style.

File: src/js/profile/mobile/widget/TextInput.js (hunk 3 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/mobile/widget/TextInput.js]: Done

File: src/js/profile/mobile/widget/TextInput.js (hunk 4 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/mobile/widget/TextInput.js:272]: Done

File: src/js/profile/mobile/widget/TextInput.js (hunk 2 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/mobile/widget/TextInput.js]: Done

File: src/css/profile/mobile/common/textinput.less (hunk 1 above)
Comment (TomaszLukawskiSam) [reply src/css/profile/mobile/common/textinput.less]: Done

**Review by HunseopJeong — APPROVED** [review]
LGTM

**Pull request merged**